src/main/Main.hs

```diff
@@ -773,7 +773,7 @@ execCmd ExecOpts {..} go@GlobalOpts{..} =
                 -- Add RTS options to arguments
                 let argsWithRts args = if null eoRtsOptions 
                             then args :: [String]
-                            else args ++ ["+RTS"] ++ eoRtsOptions ++ ["-RTS"]
+                            else args ++ ["+RTS"] ++ concatMap words eoRtsOptions ++ ["-RTS"]
```

Comment:
Sorry, I should have clarified that there's support in Stack already for parsing these kinds of arguments, please see: https://github.com/commercialhaskell/stack/blob/master/src/Options/Applicative/Args.hs. I _think_ `argsOption` will handle this correctly, including issues like escape characters and quoted whitespace.